### DISCUSSION TITLE
Docker: Getting "Virtual Void" when launching newer versions of Authoritative Server

### DISCUSSION THREAD

**Discussion by TimothyGaray** [PowerDNS/pdns#13545] [Q&A] [answered]
- Program: Authoritative, Recursor, dnsdist 
 - Issue type: Bug report

### Short description
Currently running pdns-auth-46:latest in docker (4.6.4).
Try and launch latest versions of 47 or 48 and keep getting the same thing:

Nov 29 14:52:46 Exiting because communicator thread died with error: virtual void GSQLBackend::getUpdatedMasters(std::vector<DomainInfo>&, std::unordered_set<DNSName>&, CatalogHashMap&) unable to retrieve list of master domains: Could not prepare statement: select d.id, d.name, d.type, d.notified_serial,d.options, d.catalog,r.content from records r join domains d on r.domain_id=d.id and r.name=d.name where r.type='SOA' and r.disabled=0 and d.type in ('MASTER', 'PRODUCER'): Unknown column 'd.options' in 'field list'
Nov 29 14:52:47 Done launching threads, ready to distribute questions

I can go back and run the 4.6.4 version and it is fine.

These are docker containers (running auth, recursor and admin).

docker run -d --name "pdns-auth" -e TZ="America/New_York" -v /mnt/data/powerdns/auth:/etc/powerdns -p "30053:53" -p "30053:53/udp" -p "8081:8081" --restart unless-stopped powerdns/pdns-auth-46:latest

After launching docker ps -a shows it continually restarting.  Only issue seen in logs is above.

I looked and it appears I should be able to upgrade from 46 or 48 but it looks like from the error that something is not right with the DB.

pdns.conf:
local-address=0.0.0.0
allow-axfr-ips=10.170.0.54/32
launch=gmysql
include-dir=/etc/powerdns/pdns.d
api-key=api_key
primary=yes
secondary=no
api=yes
webserver=yes
webserver-address=0.0.0.0
webserver-allow-from=0.0.0.0/0
webserver-password=secret
version-string=anonymous
default-ttl=1500
log-dns-queries=yes
loglevel=4
log-dns-details=no
gmysql-host=10.170.0.53
gmysql-password=secret
gmysql-dbname=powerdns
gmysql-user=powerdns_user

I was able to update to latest pdns recursor 49 with no issues.

**Comment by kpfleming** [comment, marked as answer]
It appears you have not applied a schema update that is needed for the newer version of pdns-auth you are trying to use.

**Reply by Habbie** [reply]
https://doc.powerdns.com/authoritative/upgrading.html

**Comment by TimothyGaray** [comment]
Yep, that was it.

I found this:  4.3.0_to_4.7.0_schema.mysql.sql

Ran it against the DB and now it starts just fine.

Thank you!
